src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerConfig.java

```diff
@@ -69,4 +70,22 @@
     super(CONFIG, props, false);
   }
 
+  /**
+   * This method won't call {@link Configurable#configure(Map)} for {@link LiKafkaProducerConfig#AUDITOR_CLASS_CONFIG}
+   * during instance creation
+   */
+  @Override
+  public <T> T getConfiguredInstance(String key, Class<T> t) {
+    if (AUDITOR_CLASS_CONFIG.equals(key)) {
```

Comment:
Not sure how you want to do this just by overriding ```getConfiguredInstance```.

Here is the method itself:
```java
public <T> T getConfiguredInstance(String key, Class<T> t) {
  Class<?> c = getClass(key);
  if (c == null)
    return null;
  Object o = Utils.newInstance(c);
  if (!t.isInstance(o))
    throw new KafkaException(c.getName() + " is not an instance of " + t.getName());
  if (o instanceof Configurable)
    ((Configurable) o).configure(originals());
  return t.cast(o);
}
```

Do you want the new code to look like this? Swap ```originals()``` with ```configsWithCurrentProducer()```. This approach won't work because we don't have ```_producerObject_``` defined in LiKafkaProducerConfig object or passed via parameter.

```java
@Override
public <T> T getConfiguredInstance(String key, Class<T> t) {
  Class<?> c = getClass(key);
  if (c == null)
    return null;
  Object o = Utils.newInstance(c);
  if (!t.isInstance(o))
    throw new KafkaException(c.getName() + " is not an instance of " + t.getName());
  if (o instanceof Configurable)
    ((Configurable) o).configure(configsWithCurrentProducer(_producerObject_));
  return t.cast(o);
}
```

There are few ways to fix this:
1. Introduce method that overloads ```getConfiguredInstance``` with extra parameter:
```public <T> T getConfiguredInstance(String key, Class<T> t, Producer<byte[], byte[]> producer)```
2. Pass producer object into LiKafkaProducerConfig using setter after it get initialized in LiKafkaProducerImpl.

I'm leaning towards the first approach. What do you think @becketqin ?